apps/verticalhome/js/app.js

```diff
@@ -150,6 +178,191 @@
+      var scrollDiff = insrertedIndex - maxDataIndex;
+      var selectedPinnedApp = app.pinnedAppsNavigation.elements[lastTimeSelected];
+
+      selectedPinnedApp.classList.remove('selected');
+
+      for(var z = 0; z < scrollDiff; z++){
+        var itm = app.pinnedAppsNavigation.elements[0];
+        var cln_tmp = itm.cloneNode(true);
+        app.pinnedAppsNavigation.elemList.appendChild(cln_tmp);
+        itm.parentNode.removeChild(itm);
+      }
+
+      app.pinnedAppsNavigation.elements[1].classList.add('selected');
+
+      app.pinnedAppsNavigation.scrollNum();
+
+      app.pinnedAppsNavigation.refresh();
+
+    },
+
+    unpinFromMainScreen: function(){
+
+      var selectedPinAppItem = document.querySelector(
+        '#pinned-apps-list .selected'
+        );
+      var selectedPinupAppItem_data_index = selectedPinAppItem
+        .getAttribute('data-index');
+
+      var notif = document.getElementById('blinknotif');
+
+      app.pinnedAppsNavigation.refresh();
+
+      this.deletedApps.push(
+        app.pinnedAppsManager.items[selectedPinupAppItem_data_index]
+        );
+
+      app.pinnedAppsManager.items[selectedPinupAppItem_data_index].
+        element.style.display = 'none';
+
+      if (selectedPinAppItem.parentNode) {
+        selectedPinAppItem.parentNode.removeChild(selectedPinAppItem);
+      }
+
+      app.pinnedAppsNavigation.elements[1].classList.add('selected');
+      app.pinnedAppsNavigation.elemList.classList.add('animation');
+      app.pinnedAppsNavigation.refresh();
+
+      this.showNotif('App removed from Homescreen and moved to "More apps"');
+
+    },
+
+    personalize: function(){
+      document.getElementById('main-screen').classList
+        .add('personalize_mode');
+
+      if( !document.getElementById('addFromMoreApps')){
+        app.pinnedAppsNavigation.reset();
+
+        var plusButton = document.createElement('div');
+        plusButton.classList.add('pinned-app-item');
+        plusButton.setAttribute('id', 'addFromMoreApps');
+        plusButton.setAttribute('data-index', '998');
+        plusButton.innerHTML = '<span style="visibility: visible;" '+
+        'class="title">+</span>';
+
+        var pinList = document.getElementById('pinned-apps-list');
+        var moreAppsLi = document.getElementById('moreApps');
+        pinList.insertBefore(plusButton, moreAppsLi);
+      }
+    },
+
+    endPersonalize: function(){
+      var plusButton = document.getElementById('addFromMoreApps');
+      document.getElementById('main-screen').classList
+        .remove('personalize_mode');
+      if (plusButton.parentNode) {
+        plusButton.parentNode.removeChild(plusButton);
+      }
+
+    },
+
+    rearrange: function(){
+      document.getElementById('main-screen').classList
+        .add('rearrange_mode');
+    },
+
+    exitRearrange: function(){
+      document.getElementById('main-screen').classList
+        .remove('rearrange_mode');
+    },
+
+    enterMoreAppsPersonalise: function(){
```

Comment:
You should use a common grammar naming variables. Current code contains both personaliSe and personaliZe names of variables